### ISSUE TITLE
Chat log/history export

### ISSUE THREAD

**Issue by jkufner** [siacs/Conversations#909]
Until message archive management become reality, chat history is stored only on clients. It would be nice to have simple option to export the full history to some parsable format (like JSON or SQLite database), so it can be easily converted and imported into other Jabber client or to future server-side message archive.

For example Gajim stores history in SQLite database, so import can be done using few SQL queries.

The export should be lossless and as simple as possible (from programmer's point of view). It can be hidden in somewhere in settings. Dump should be stored in file on SD card with timestamp in the filename, and file share menu may be presented to user when export is complete.

**Comment by Natanji** [comment]
I would like some feature like this as well, but I'm not too certain about the best course of action. Importing into other messengers isn't easy because there is a tremendous lot of them, all with different storage formats. Emulating all of these isn't really viable, nor does it seem logical to have such a feature if only Gajim or some other messenger is supported then.

Perhaps we should *only* have an option to export them to a text file instead, in an own plaintext log format that is still somewhat easy to parse? As in, really well human-readable but still importable into other clients?

**Comment by knoy** [comment]
This kind of goes against a central tenet of "Off-The-Record" messaging, in that the conversation is explicitly supposed to be destroyed afterwards.

**Comment by SamWhited** [comment]
Conversations (and most clients) don't destroy OTR chats by default. OTR has nothing to do with what happens afterwards, it just protects the messages in transit. The end parties (once verified) are assumed to be trusted (which includes handling the information in whatever way they see fit).

**Comment by jkufner** [comment]
Any plain text file is hard to parse. Nicely formatted JSON is easy to
read by human and easy to parse by machine (import script).
Alternatively it can be XML file with some basic XSLT template to make
it readable using any web browser.

**Comment by iNPUTmice** [comment]
> Until message archive management become reality

MAM is actually a reality right now. It has been in Conversations now for several releases and is working great.

**Comment by STPKITT** [comment]
MAM doesn't help when you just want your chat log/history exported to a neutral txt-/json-/xml-/etc.-file, so chat log/history export is a highly sought after feature for me and some of my friends and I brought that topic up on the Converversations support chat group and had immediately answer from another user who wished the same feature. Google XMPP-Server let's you download your chat-log in a not really human-reabable json file (for example contact-names are some numbercode) and for that reason me and my friends still use Google XMPP despite it being unsecure and blocked by several other XMPP-server-providers.

**Comment by sampablokuper** [comment]
+1 for export. Please re-open this feature request.

**Comment by sbrl** [comment]
I'd love to be able to export the chat history so that I can read it elsewhere. Even just a text file would be great!

**Comment by iNPUTmice** [comment]
> I'd love to be able to export the chat history so that I can read it elsewhere. Even just a text file would be great!

That has been possible for a while now. You can find the button in the expert settings.

**Comment by sbrl** [comment]
@iNPUTmice Oh awesome, thanks! I didn't see that there.

**Comment by sbrl** [comment]
Erm hang on, @iNPUTmice - where does it export to?

**Comment by iNPUTmice** [comment]
Onto the SD card in a directory called Conversations/logs/

**Comment by sbrl** [comment]
Awesome, thanks!

**Comment by arkap** [comment]
> > I'd love to be able to export the chat history so that I can read it elsewhere. Even just a text file would be great!
> 
> That has been possible for a while now. You can find the button in the expert settings.

Just stumbled over this looking for a way to export a chat log to a file. Has the option been removed from expert settings? Is there another way?

**Comment by licaon-kter** [comment]
@arkap removed a long time ago, when Backup/Restore was added.

So, make a Backup (you do know your account password, yes?) then use https://github.com/iNPUTmice/ceb2txt or https://gitlab.com/alping/ceb-extract to extract the text.

**Comment by arkap** [comment]
> @arkap removed a long time ago, when Backup/Restore was added.
> 
> So, make a Backup (you do know your account password, yes?) then use https://github.com/iNPUTmice/ceb2txt or https://gitlab.com/alping/ceb-extract to extract the text.

That'll work, thanks!
